controllers/datadogagent/feature/cspm/feature.go

```diff
@@ -0,0 +1,232 @@
+// It should do nothing if the feature doesn't need to configure it.
+func (f *cspmFeature) ManageClusterAgent(managers feature.PodTemplateManagers) error {
+	cmVol, cmVolMount := volume.GetConfigMapVolumes(
+		f.configMapConfig,
+		f.configMapName,
+		cspmConfigVolumeName,
+		cspmConfigVolumePath,
+	)
+	managers.Volume().AddVolumeToContainer(&cmVol, &cmVolMount, apicommonv1.ClusterAgentContainerName)
+
+	enabledEnvVar := &corev1.EnvVar{
+		Name:  apicommon.DDComplianceEnabled,
+		Value: "true",
+	}
+	managers.EnvVar().AddEnvVarToContainer(apicommonv1.ClusterAgentContainerName, enabledEnvVar)
+
+	if f.checkInterval != "" {
+		intervalEnvVar := &corev1.EnvVar{
+			Name:  apicommon.DDComplianceCheckInterval,
+			Value: f.checkInterval,
+		}
+		managers.EnvVar().AddEnvVarToContainer(apicommonv1.ClusterAgentContainerName, intervalEnvVar)
+	}
+
+	return nil
+}
+
+// ManageNodeAgent allows a feature to configure the Node Agent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *cspmFeature) ManageNodeAgent(managers feature.PodTemplateManagers) error {
+	capabilities := []corev1.Capability{
+		"AUDIT_CONTROL",
+		"AUDIT_READ",
+	}
+	managers.SecurityContext().AddCapabilitiesToContainer(capabilities, apicommonv1.SecurityAgentContainerName)
+
+	// configmap volume mount
+	cmVol, cmVolMount := volume.GetConfigMapVolumes(
+		f.configMapConfig,
+		f.configMapName,
+		cspmConfigVolumeName,
+		cspmConfigVolumePath,
+	)
+	managers.Volume().AddVolumeToContainer(&cmVol, &cmVolMount, apicommonv1.SecurityAgentContainerName)
+
+	// cgroups volume mount
+	cgroupsVol, cgroupsVolMount := volume.GetVolumes(apicommon.CgroupsVolumeName, apicommon.CgroupsHostPath, apicommon.CgroupsMountPath, true)
+	managers.Volume().AddVolumeToContainer(&cgroupsVol, &cgroupsVolMount, apicommonv1.SecurityAgentContainerName)
+
+	// passwd volume mount
+	passwdVol, passwdVolMount := volume.GetVolumes(apicommon.PasswdVolumeName, apicommon.PasswdHostPath, apicommon.PasswdMountPath, true)
+	managers.Volume().AddVolumeToContainer(&passwdVol, &passwdVolMount, apicommonv1.SecurityAgentContainerName)
+
+	// procdir volume mount
+	procdirVol, procdirVolMount := volume.GetVolumes(apicommon.ProcdirVolumeName, apicommon.ProcdirHostPath, apicommon.ProcdirMountPath, true)
+	managers.Volume().AddVolumeToContainer(&procdirVol, &procdirVolMount, apicommonv1.SecurityAgentContainerName)
+
+	// host root volume mount
+	hostRootVol, hostRootVolMount := volume.GetVolumes(apicommon.HostRootVolumeName, apicommon.HostRootHostPath, apicommon.HostRootMountPath, true)
+	managers.Volume().AddVolumeToContainer(&hostRootVol, &hostRootVolMount, apicommonv1.SecurityAgentContainerName)
+
+	// group volume mount
+	groupVol, groupVolMount := volume.GetVolumes(apicommon.GroupVolumeName, apicommon.GroupHostPath, apicommon.GroupMountPath, true)
+	managers.Volume().AddVolumeToContainer(&groupVol, &groupVolMount, apicommonv1.SecurityAgentContainerName)
+
+	// env vars
```

Comment:
Do we need the `HOST_ROOT` env var for this or is that automatically added elsewhere?